src/CmsBundle/Security/ContentVoter.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Opifer\CmsBundle\Security;
+
+use Opifer\CmsBundle\Entity\User;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Component\Security\Core\Security;
+
+class ContentVoter extends Voter
+{
+    private $security;
+    private $container;
+    private $roles;
+
+    public function __construct(Security $security, ContainerInterface $container, $roles)
+    {
+        $this->security = $security;
+        $this->container= $container;
+        $this->roles = $roles;
+    }
+
+    protected function voteOnAttribute($attribute, $subject, TokenInterface $token)
+    {
+        $user = $token->getUser();
+
+        if (!$user instanceof User) {
+            // the user must be logged in; if not, deny access
+            return false;
+        }
+
+        foreach($user->getRoles() as $role) {
+            if (in_array($attribute, $attribute['content_roles'])) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    protected function supports($attribute, $subject)
+    {
+        return (isset($attribute['content_roles']) && is_array($attribute['content_roles']));
```

Comment:
`$attribute` should be a simple string (The key you're checking permission for. E.g. `CONTENT_EDIT`). The `$subject` should hold the `Content` object.

You can then simply check if the `$subject` is an instance of `Content` and if it has any roles. If so, return `true`